### ISSUE TITLE
plotPartialDependence: bug with factor input

### ISSUE THREAD

**Issue by berndbischl** [mlr-org/mlr#1174]
```
task = makeRegrTask(data = iris, target = "Sepal.Length")
lrn = makeLearner("regr.lm")
mod = train(lrn, task)
pd1 = generatePartialDependenceData(mod, task)
print(pd1)
print(plotPartialDependence(pd1))


pd2 = generatePartialDependenceData(mod, task, features = getTaskFeatureNames(task)[4])
print(pd2)
print(plotPartialDependence(pd2))
```

pd1 is totally broken, hence the plot of course too
```
PartialDependenceData
Task: iris
Features: Sepal.Width, Petal.Length, Petal.Width, Species
Target: Sepal.Length
Derivative: FALSE
Interaction: FALSE
Individual: FALSE
  Sepal.Length Sepal.Width Petal.Length Petal.Width Species
1        5.319       2.000           NA          NA    <NA>
2        5.451       2.267           NA          NA    <NA>
3        5.583       2.533           NA          NA    <NA>
4        5.716       2.800           NA          NA    <NA>
5        5.848       3.067           NA          NA    <NA>
6        5.980       3.333           NA          NA    <NA>
... (33 rows, 5 cols)

```

pd2 looks better, but the plot is completely empty
```
PartialDependenceData
Task: iris
Features: Species
Target: Sepal.Length
Derivative: FALSE
Interaction: FALSE
Individual: FALSE
  Sepal.Length    Species
1        6.426     setosa
2        5.702 versicolor
3        5.402  virginica
```

**Comment by zmjones** [comment]
this is a weird case no? you are doing regression on multilevel factor?

**Comment by berndbischl** [comment]
> this is a weird case no? you are doing regression on multilevel factor?

i dont understand that sentence? did you misread something? the factor is simply an input feature here...

**Comment by zmjones** [comment]
the generation function output isn't broken at all as far as i can see. in the first case the data element of the container has a column for each of the features and the target, and since interaction = FALSE the column of each feature is NA except for gridsize rows, unless there are fewer than gridsize unique values in the input data, as is the case with the Species feature. 

The plot is messed up though. Checking that out now.

**Comment by zmjones** [comment]
i just saw that you were using the iris data and thought at first you were using regression on that. it was early in the morning...

**Comment by berndbischl** [comment]
> the generation function output isn't broken at all as far as i can see.

ah i always get confused by this weird "rbind.fill-format"....
i looked at pd1 again, and it seems ok. (which is kinda bad, as now we cannot unit test the problems)

so then both plots for pd1 and pd2 are broken.

**Comment by zmjones** [comment]
just issued a pr to fix it.

when we switched to data.table instead of plyr/reshape2 we started to use their melt function as well. if you gave reshape2::melt.data.frame measure.vars of mixed type it would cast everything to a numeric with a warning. data.table makes everything a character instead, which broke the plot. i added an explicit cast to numeric for factors now. 

of course at some level this isn't how you want to handle factor plotting at all. what you'd want is no line for unordered factor and axis labels that match levels. we can't do this with facetting though. all the x-variables have to be of the same type. so i think this casting is the least bad alternative.
